api/src/main/java/com.redhat.devtools.recognizer.api/LanguageRecognizer.java

```diff
@@ -10,9 +10,12 @@
  ******************************************************************************/
 package com.redhat.devtools.recognizer.api;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.List;
 
 public interface LanguageRecognizer {
+    String selectDevFile(String srcPath, List<DevfileType> devfileTypes, List<File> devfiles) throws IOException;
+    String selectDevFileFromTypes(String srcPath, List<DevfileType> devfileTypes) throws IOException;
```

Comment:
Should return one of the input objects. And maybe we need an intermediate object as we may select several devfiles so we should be able to return an ordered list and let the user decide which one to pick. Also the first method needs to load all devfiles from registry so I would replace the second parameter by a DevfileResolver that take a DevfileType and returns a Reader. So that make me think that for odo we need to have the registry name somewhere to DevfileType should be an interface.
Also suggesting getDevfile as a better method name